### DISCUSSION TITLE
Block-based outliner?

### DISCUSSION THREAD

**Discussion by drawtheliner** [AppFlowy-IO/AppFlowy#1957] [General]
I just installed Appflowy. Looks good! I missed a few key features I'd expect in this type of software though, the main ones being:
- Block granularity: have blocks (paragraphs and other block types) be the smallest unit of content, which can be moved around, queried and referenced individually
- Outliner features: ability to quickly create nested outlines, fold/unfold items, move things around, all via keyboard (right now it seems a Page is very much a linear sequence of paragraphs)
- Reference pages/blocks inline
- Transclusion of blocks
- Be able to run queries and display as a grid, table, calendar view
- Have blocks be able to have properties (which could then be used for queries and displaying them)

Are these planned?

**Comment by annieappflowy** [comment]
Hi @drawtheliner , very much appreciate your input.
Many features you mentioned are planned to develop in the near term. You can check out our short-term (6-month) [roadmap ](https://github.com/AppFlowy-IO/AppFlowy/discussions/1715) to see what's on the radar. And you can follow our [Twitter](https://twitter.com/appflowy) or the [announcement](https://discord.gg/aAGATKfwVn) channel in the Discord server to get the latest update.

Could you please elaborate on this --> Have blocks be able to have properties?

Thanks,
Annie

**Reply by drawtheliner** [reply]
Hi @annieappflowy thanks for that.

I couldn't find any of the items I mentioned in the roadmap you linked. Is any of it planned at all?
